src/main/webapp/app/overview/course-lectures/course-lecture-details.component.ts

```diff
@@ -51,6 +53,9 @@ export class CourseLectureDetailsComponent implements OnInit, OnDestroy {
                 this.lecture = null;
                 this.lectureService.find(params.lectureId).subscribe((lectureResponse: HttpResponse<Lecture>) => {
                     this.lecture = lectureResponse.body;
+                    if (this.studentQuestions) {
+                        this.onChildActivate(this.studentQuestions);
```

Comment:
can you document this? I'm not really sure if I understand the code.
If this.studentQuestions is not undefined and not null, which part has injected it?
As it is not a ViewChild, I don't see the magic behind this approach

Reply:
I updated the code to be more clear.
onChildActivate gets called by the router outlet when the child component is mounted. If the exercise is already loaded it gets set to the mounted component. But it might also be that the component gets mounted before the exercise is fetched therefore I save a reference to the component instance and also set the exercise when it is loaded. So it works both ways whichever gets loaded first.